src/module/item/spell/index.ts

```diff
@@ -261,6 +304,33 @@ class SpellPF2e extends ItemPF2e {
             .sort((first, second) => first.level - second.level);
     }
 
+    /** Returns a specific variant of this spell or a shows a prompt to select one. */
+    async getVariantChoices(this: Embedded<SpellPF2e>, overlayId?: string): Promise<Embedded<SpellPF2e> | null> {
```

Comment:
Might need a better name, this doesn't return the variant choices.

@stwlam can you think of something? I'm thinking of getOrPromptVariant().

Reply:
I'll strip out the `overlayId` param as that would now only work with an override overlay and rename it to `promptVariant()`.
The `overlayIds[] -> spellVariant` concept needs a way to specifiy which overlayIds represent a single spell variant.
When that is done the method could be called with the `variantId` again.